src/functions.cpp

```diff
@@ -2364,7 +2364,58 @@ RCP<const Basic> acoth(const RCP<const Basic> &arg)
 
 RCP<const Basic> ACoth::diff(const RCP<const Symbol> &x) const
 {
-    return mul(div(one, sub(one, pow(x, i2))), get_arg()->diff(x));
+    return mul(div(one, sub(one, pow(get_arg(), i2))), get_arg()->diff(x));
```

Comment:
Ah, we should add a new test that is not `acoth(x)` or `acoth(-x)` in a new PR when other inverse hyperbolic functions are fixed